编译错误 mkldnn/mul_mkldnn_op.cc:84:5

/home/kgdev/Paddle/paddle/fluid/operators/mkldnn/mul_mkldnn_op.cc:84:5: error: operation on ‘dst_fmt’ may be undefined [-Werror=sequence-point]

dst_fmt != (dst_fmt = memory::format::ncdhw)) should it be src_fmt != (dst_fmt = memory::format::ncdhw)) ???

--- a/paddle/fluid/operators/mkldnn/mul_mkldnn_op.cc
+++ b/paddle/fluid/operators/mkldnn/mul_mkldnn_op.cc
@@ -84,7 +84,7 @@ class MulPrimitiveFactory {
     if ((data->dims().size() == 4 &&
          src_fmt != (dst_fmt = memory::format::nchw)) ||
         (data->dims().size() == 5 &&
-         dst_fmt != (dst_fmt = memory::format::ncdhw))) {
+         src_fmt != (dst_fmt = memory::format::ncdhw))) {
       auto dst_mdesc = CreateMemDescriptor<T>(data, dst_fmt);
       x_tmp.mutable_data<T>(ctx.GetPlace(), data->memory_size());
 

